src/Microsoft.OData.Core/UriParser/SemanticAst/ODataPath.cs

```diff
@@ -22,10 +22,10 @@ namespace Microsoft.OData.UriParser
     [SuppressMessage("Microsoft.Naming", "CA1710:IdentifiersShouldHaveCorrectSuffix", Justification = "ODataPathCollection just doesn't sound right")]
     public class ODataPath : IEnumerable<ODataPathSegment>
     {
-        /// <summary>
-        /// The segments that make up this path.
-        /// </summary>
-        private readonly IList<ODataPathSegment> segments;
+        /// <summary>	
+        /// The segments that make up this path.	
+        /// </summary>	
+        private readonly List<ODataPathSegment> segments;

```

Comment:
Why did you replace the IList with a concrete type?

Reply:
List exposes extra methods that are more performant which is something I am relying on.

Reply:
#Resolved